frontends/web/src/components/aopp/aopp.tsx

```diff
@@ -34,85 +40,189 @@ interface SubscribedProps {
+                        <AppLogo />
+                        <Header title={t('aopp.title')} />
+                        <Content>
+                            <p>{t('aopp.addressRequest', { host: domain })}</p>
+                        </Content>
+                        <FullscreenButtons>
+                            <Button primary onClick={aoppAPI.approve}>{t('button.continue')}</Button>
+                            <Button secondary onClick={aoppAPI.cancel}>{t('dialog.cancel')}</Button>
+                        </FullscreenButtons>
+                    </Fullscreen>
                 );
-
             case 'awaiting-keystore':
                 return (
-                    <div>
-                        <p>{ t('aopp.addressRequested', { host: aopp.callback }) }</p>
-                        <p>{ t('aopp.awaitingKeystore') }</p>
-                        <Button primary onclick={aoppAPI.cancel}>Cancel</Button>
-                    </div>
+                    <Banner>{t('aopp.banner')}</Banner>
                 );
-            case 'choosing-account': {
-                const options = aopp.accounts.map(account => {
-                    return {
-                        text: account.name,
-                        value: account.code,
-                    };
-                });
+            case 'choosing-account':
+                if (aopp.accounts) {
+                    const options = aopp.accounts.map(account => {
+                        return {
+                            text: account.name,
+                            value: account.code,
+                        };
+                    });
+                    return (
+                        <Fullscreen>
+                            <Header title={t('aopp.title')}>
+                                <p>{domain}</p>
+                            </Header>
+                            <form onSubmit={this.chooseAccount}>
+                                <Content>
+                                    <Select
+                                        label={t('buy.info.selectLabel')}
+                                        options={options}
+                                        defaultValue={options[0].value}
+                                        value={accountCode}
+                                        onChange={e => this.setState({ accountCode: e.target.value })}
+                                        id="account" />
+                                </Content>
+                                <FullscreenButtons>
+                                    <Button primary type="submit">{t('button.next')}</Button>
+                                    <Button secondary onClick={aoppAPI.cancel}>{t('dialog.cancel')}</Button>
+                                </FullscreenButtons>
+                            </form>
+                        </Fullscreen>
+                    );
+                }
+                // show syncing message when no aopp.accounts are found
```

Comment:
if the state is choosing-accounts, there is always at least one account, otherwise the backend would jump to the error state.

maybe there is a way to encode it in the types, but otherwise feel free to just `!` or `?` it or whatever the syntax is to override typescript with your better knowledge :)

Reply:
I don't like https://github.com/digitalbitbox/bitbox-wallet-app/pull/1373/commits/07dd8d7382b5daf0be1ecbdc722f932b4133c2d2

will look if there is an easy nicer solution in the interface.

Reply:
fixed in https://github.com/digitalbitbox/bitbox-wallet-app/compare/07dd8d7382b5daf0be1ecbdc722f932b4133c2d2..a11ea8a3885e5e2a801d19718ae46d6bfb0b5d2b